pkg/gadget-service/service.go

```diff
@@ -126,9 +151,12 @@ func (s *Service) RunGadget(runGadget pb.GadgetManager_RunGadgetServer) error {
 		return fmt.Errorf("setting parameters: %w", err)
 	}
 
-	if c, ok := gadgetDesc.(gadgets.GadgetDescCustomParser); ok {
-		var err error
-		parser, err = c.CustomParser(gadgetParams, request.Args)
+	if c, ok := gadgetDesc.(runTypes.RunGadgetDesc); ok {
+		gadgetInfo, err := s.runtime.GetGadgetInfo(gadgetDesc, gadgetParams, request.Args)
+		if err != nil {
+			return fmt.Errorf("getting gadget info: %w", err)
+		}
+		parser, err = c.CustomParser(gadgetInfo)
```

Comment:
Maybe I got a little bit lost, but isn't it strange that the gadget-service, which runs on the gadget pod side, needs the parser? It seems the parser is implementing too many things, don't you think?

Reply:
The "parser" isn't a parser anymore, it's much more than this. We need to find for a different name for this, but it's not related to this PR. cc @flyth